### ISSUE TITLE
Research Pull Request Triggers in GitHub

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#50794]
# Description
How might we trigger a workflow when a push event occurs for a branch with a pull request?

# Tasks
- [ ] Read through GitHub Actions's documentation to determine if there is a manner in which an event can be specified only for when a user pushes code to a branch with a pull request created
- [ ] Define an approach for triggering a workflow using GHA capabilities

# Acceptance Criteria
- [ ] The deployment workflow is triggered by a push even occurring against a branch with an existing pull request

**Comment by CBonade** [comment]
GitHub has come a long way in the triggers available for pull request since its' beta days. The following declaration for workflow trigger will serve our use case here:

`on:
  pull_request:
    types: [opened, synchronize, ready_for_review]
`

These are defined as following:

`opened`: This will trigger the workflow when a new PR is opened.
`synchronize`: This is an event that fires when code is pushed to an open PR and the PR synchronizes with the updated version of the branch. This would ensure that pushes prior to the PR being open do not trigger the workflow, only after.
`ready_for_review`: this covers a drafted PR that was already opened some time ago, being set as ready to be a public PR.

I've tested these triggers on a test branch with a sample workflow and they all work as expected. All that remains is we need to decide if these triggers are preferable for our workflows to the push ones currently in place.

[Supporting Documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)

**Comment by pjhill** [comment]
Awesome, sounds like we have a way forward with triggering workflows for the exact events that we need!
